tests/Tests.hs

```diff
@@ -50,6 +53,28 @@ prop_split_string ls =
   not (null ls) ==>
     splitWithSeparator '/' (T.pack ls) == T.split (=='/') (T.pack ls)
 
+instance Arbitrary TestDwarfUUID where
+  arbitrary = do
+    uuid <- arbitraryUUID
+    arch <- arbitraryArch
+    return $ TDUUID (toInputLine uuid arch) uuid arch
+    where
+      toInputLine uuid arch =
+        "UUID: " ++ uuid ++ " (" ++ show arch ++ ") Carthage/Build/iOS/Foo.framework/Foo"
+      arbitraryUUID = fmap (intercalate "-")
+                           (sequence [vectorOf 8 hexDigits, vectorOf 4 hexDigits, vectorOf 4 hexDigits, vectorOf 12 hexDigits])
+      hexDigits = elements (concat [['A'..'F'], ['0'..'9']])
```

Comment:
@blender the reason why the test didn't work as we looked at it was because this was defined as: `hexDigits = elements (concat [['A'..'Z'], ['0'..'9']])` that is I completely missed that UUIDs where hexadecimals! 😭 So parsing was failing and I suppose that quickcheck was trying to shrink the test but didn't manage to find a minimal test case.

Reply:
Was not expecting that one 🤣